explainer.md

```diff
@@ -404,178 +437,110 @@ There are many examples like this for the web: Google Sheets, Figma's plugins, o
+  return globalThis[property];
 }
 ```
 
-### <a name='Example:SimpleSubclass'></a>Example: Simple Subclass
-
-```js
-class EmptyRealm extends Realm {
-  constructor(...args) {
-    super(...args);
-    let globalThis = this.globalThis;
-
-    // delete global descriptors:
-    delete globalThis.Math;
-    ...
-  }
-}
-```
-
-In the example above, the global object of a newly created realm of `EmptyRealm` will have no global `Math` available.
-
-### <a name='Example:DOMMocking'></a>Example: DOM Mocking
-
-```js
-const realm = new Realm();
-
-await installFakeDOM(realm.globalThis);
-
-// Custom properties are now added to the Realm such as
-realm.globalThis.document;
-realm.globalThis.top;
-realm.globalThis.location;
-```
-
 ### <a name='Example:iframesvsRealms'></a>Example: iframes vs Realms
 
-If you're using anonymous iframes today to "evaluate" javascript code in a different realm, you can replace it with a new Realm, as a more performant option, e.g.:
+If you're using anonymous iframes today to "evaluate" javascript code in a different realm, you can replace it with a new Realm, as a more performant option, without identity discontinuity, e.g.:
 
-```js
+```javascript
 const globalOne = window;
 let iframe = document.createElement('iframe');
 document.body.appendChild(iframe);
-const globalTwo = iframe.contentWindow;
-```
-
-will become:
-
-```js
-const globalOne = window;
-const globalTwo = new Realm().globalThis;
-```
+const iframeArray = iframe.contentWindow.Array;
 
-### <a name='Example:IndirectEvaluation'></a>Example: Indirect Evaluation
+console.assert(iframeArray !== Array);
 
-This operation should be equivalent, in both scenarios:
+const list = iframeArray('a', 'b', 'c');
 
-```js
-globalOne.eval('1 + 2'); // yield 3
-globalTwo.eval('1 + 2'); // yield 3
+list instanceof Array; // false
+[] instanceof iframeArray; // false
+Array.isArray(list); // true
 ```
 
-### <a name='Example:DirectEvaluation'></a>Example: Direct Evaluation
-
-This operation should be equivalent, in both scenarios:
-
-```js
-globalOne.eval('eval("1 + 2")'); // yield 3
-globalTwo.eval('eval("1 + 2")'); // yield 3
-```
-
-### <a name='Example:IdentityDiscontinuity'></a>Example: Identity Discontinuity
-
-Considering that you're creating a brand new realm, with its brand new global variable,
-the identity discontinuity is still present, just like in the iframe example:
-
-```js
-let a1 = globalOne.eval('[1,2,3]');
-let a2 = globalTwo.eval('[1,2,3]');
-a1.prototype === a2.prototype; // yield false
-a1 instanceof globalTwo.Array; // yield false
-a2 instanceof globalOne.Array; // yield false
-```
+This code is **not** possible with the Realms API! Non-primitive values are not transfered cross-realms using the Realms API.
 
 ## <a name='Example:NodesvmobjectsvsRealms'></a>Example: Node's vm objects vs Realms
 
 If you're using node's `vm` module today to "evaluate" javascript code in a different realm, you can replace it with a new Realm, e.g.:
```

Comment:
This is no longer quite true, given the restrictions. You probably want to explain that.

Reply:
246e41b

It's not much but it provides some honest information. WDYT, @littledan?